src/components/core/text-field/text-field.tsx

```diff
@@ -0,0 +1,972 @@
+import React from 'react';
+import Button from '@core/button/button';
+import Badge from '@core/badge/badge';
+import CircularCloseIcon from '@assets/svg/circular-close-icon.svg';
+import { forwardRef, Fragment, InputHTMLAttributes, ReactNode, useState, useRef, useEffect } from 'react';
+import { createCanvas, getCanvasFont } from './utils';
+import { styled } from 'Styles/stitches.config';
+
+type InputTypes = 'text' | 'number' | 'email' | 'password' | 'tel' | 'textarea';
+type TWordCountProps = { count: number; max_length: number };
+type TPasswordStrengthProps = { dark: boolean; disable_meter: boolean; user_input: string };
+type THintTextProps = { error: string; hint: string; success: string };
+type TBadge = {
+    id: string;
+    label: string;
+};
+type TAlignValue = 'left' | 'center' | 'right';
+
+export type TextFieldProps = InputHTMLAttributes<HTMLInputElement | HTMLTextAreaElement> & {
+    align_value?: TAlignValue;
+    button_label?: string;
+    currency_suffix_element?: string;
+    dark?: boolean;
+    default_badges: TBadge[];
+    has_password_meter?: boolean;
+    hint_text?: THintTextProps;
+    inline_prefix_element?: ReactNode;
+    inline_suffix_element?: ReactNode;
+    is_borderless?: boolean;
+    label?: string;
+    max_length?: number;
+    number_of_badges?: number;
+    prefix_style?: { [key: string]: string };
+    suffix_style?: { [key: string]: string };
+    type?: InputTypes;
+    with_badges?: boolean;
+    onBadges?: (badges: TBadge[]) => void;
```

Comment:
What is this for? onBadges event naming is weird and don't give any clarity on when the event is fired

Reply:
@yashim-deriv `onBadges` function is fired when the user adds new badge in input field. I have added this callback to have a possibility to save badges outside the component (e.g in state or store), that when i need to populate TextField with badges i can provide an array with saved badges. Should I change the name of the callback in this case?

Reply:
`onAddBadge` perhaps?